crates/tick/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#![doc(html_logo_url = "https://media.githubusercontent.com/media/microsoft/oxidizer/refs/heads/main/crates/tick/logo.png")]
+#![doc(html_favicon_url = "https://media.githubusercontent.com/media/microsoft/oxidizer/refs/heads/main/crates/tick/favicon.ico")]
+#![cfg_attr(
+    test,
+    allow(
+        clippy::arithmetic_side_effects,
+        clippy::unchecked_duration_subtraction,
+        reason = "allow these lints in tests to improve the readability of the tests"
+    )
+)]
+#![cfg_attr(coverage_nightly, feature(coverage_attribute))]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+
+//! Runtime-agnostic primitives for obtaining, working with, and mocking system
+//! time and timers, enabling faster and more robust testing.
+//!
+//! # Quick Start
+//!
+//! ```no_run
+//! use std::time::Duration;
+//! use tick::{Clock, Delay};
+//!
+//! #[tokio::main]
+//! async fn main() {
+//!     // Create a clock using the Tokio runtime
+//!     let clock = Clock::new_tokio();
+//!
+//!     // Get the current timestamp
+//!     let now = clock.timestamp();
+//!     println!("Current time: {now}");
+//!
+//!     // Delay execution
+//!     Delay::new(&clock, Duration::from_secs(1)).await;
```

Comment:
Have you consider this syntax instead:

```rust
clock.delay(Duration::from_secs(1)).await;
```

A bit shorter and perhaps more discoverable?

Reply:
Hmmm. In terms of being idiomatic, to me this lines up with the builder pattern somewhat.

When you have Foo and FooBuilder, we don't expose Foo::new and FooBuilder::new, but instead we expose Foo::builder.

So to me, Clock::delay is like Foo::builder...

Reply:
I'm not very opposed to it. 

The main argument against it, as a general pattern, is that it muddles responsibilities. Like why might `clock.delay()` exist and not `clock.something_else()` for an arbitrary `SomethingElse` that happens to want a clock? We therefore wanted to be conservative adding these, unless there was a clear UX benefit.

That said, I've found myself typing a few times `Sleep`?... `Timeout`? ... `Delay`; and `clock.delay()` might be one of these cases.